File: C/interview_preparation_kit/arrays/problem_3_New_Year_Chaos.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void minimumBribes(int q_count, int *q) {
  int bribes = 0;
  for (int i = q_count - 1; i >= 0; i--) {
    if (q[i] - (i + 1) > 2) {
      printf("Too chaotic\n");
      return;
    }
    for (int j = q[i] - 2 > 0 ? q[i] - 2 : 0; j < i; j++) {
      if (q[j] > q[i]) {
        bribes++;
      }
    }
  }
  printf("%d\n", bribes);
}

int main() {
  int q_count = 8;
  int q[8] = {5, 1, 2, 3, 7, 8, 6, 4};
  clock_t start = clock();
  minimumBribes(q_count, q);
  clock_t end = clock();

  double execution_time = (double)(end - start) / CLOCKS_PER_SEC;
  printf("Execution Time: %f seconds\n", execution_time);
  return 0;
}

File: Rust/interview_preparation_kit/arrays/problem_3_New_Year_Chaos.rs
use std::time::Instant;

fn minimum_bribes(q:&[i32]) {
    let mut bribes = 0;
    for i in (0..q.len()).rev() {
        let pos = i + 1;
        let expected = pos as i32;
        let actual = q[i];
        if actual - expected > 2 {
            println!("Too chaotic");
            return;
        }
        let start = if actual - 2 > 0 { actual - 2 } else { 0 };
        let end = i as i32;
        for j in start..end {
            if q[j as usize] > actual {
                bribes += 1;
            }
        }
    }
    println!("{}", bribes);
}

fn main() {
    let q = [2, 1, 5, 3, 4];
    let start = Instant::now();
    minimum_bribes(&q);
    println!("Elapsed time: {:?}", start.elapsed());
}
